app/models/product_for_cart.rb

```diff
@@ -0,0 +1,118 @@
+class ProductForCart
+
+  include TextHelpers::Translation
+
+  attr_accessor :error_message
+  attr_accessor :error_path
+
+  def initialize(product)
+    @product = product
+  end
+
+  def purchasable_by?(acting_user, session_user)
+    error = checks(acting_user, session_user).detect do |check|
```

Comment:
Could this be `all?` instead of `detect`? Then you shouldn't need the `error.nil?` check at the end.

Reply:
`all?` will run all of the checks, and the current behavior (which we don’t want to change here) is to stop after the first encountered problem, so unfortunately, that’s why we can’t use it.

Reply:
I'm pretty sure that `all?` short circuit on the first `false` in the block. I've run into that where I _do_ want to run all checks in an array but return the `all?`.

Reply:
I didn’t understand what Jason had meant originally here, and I asked about him directly. The goal was to get rid of the local variable, which  I refactored away in 11d0f18